ci/ansible/playbooks/os/redhat/upgrade_release.yml

```diff
@@ -0,0 +1,378 @@
+        non_cloud:
+          - name: rhel-7-server-extras-rpms
+            repo_file: /etc/yum.repos.d/redhat.repo
+
+    reboot:
+      connect_timeout: 5 # maximum seconds to wait for a successful connection to the managed hosts before next trial
+      update_timeout: 500
+      upgrade_timeout: 3600 # Long wait required by Leapp utility service
+
+  tasks:
+    ### PRE-UPGRADE ###
+    - name: Assert initial major version
+      assert:
+        that: ansible_distribution_major_version == "7"
+        fail_msg: "Release upgrade not possible in current major version"
+        quiet: true
+
+    - name: Detect and set provider type
+      block:
+        - name: Register presence of distinctive AWS repository file
+          stat:
+            path: "{{ leapp_dependencies.repos.aws.0.repo_file }}"
+          register: aws_repo
+
+        - name: Register presence of distinctive Azure repository file
+          stat:
+            path: "{{ leapp_dependencies.repos.azure.0.repo_file }}"
+          register: azure_repo
+
+        - name: Set provider
+          set_fact:
+            provider: "{% if aws_repo.stat.exists %}aws{% elif azure_repo.stat.exists %}azure{% else %}non_cloud{% endif %}"
+
+    - name: Ensure Redhat subcription on non-cloud hosted systems
+      when: provider == 'non_cloud'
+      include_tasks: register_subscription.yml
+
+    - name: Register SELinux state
+      shell: "sestatus|awk '{ $NF }'"
+      register: pre_upgrade_selinux_status
+
+    - name: Register SELinux mode
+      command: 'getenforce'
+      register: pre_upgrade_selinux_mode
+
+    - name: Register active repositories
+      command: "yum repolist"
+      register: pre_upgrade_active_repositories
+
+    - name: Ensure repositories that provide leapp utility are enabled
+      ini_file:
+        path: "{{ item.repo_file }}"
+        section: "{{ item.name }}"
+        option: enabled
+        value: 1
+      loop: "{{ leapp_dependencies.repos[provider] }}"
+
+    - &UPDATE_PACKAGES
+      name: Update all packages in current major version
+      yum:
+        update_cache: true
+        name: "*"
+        state: latest
+
+    - name: Reboot system # to load kernel from latest minor version if any
+      reboot:
+        msg: Reboot initiated by Ansible due to release upgrade
+        connect_timeout: "{{ reboot.connect_timeout }}"
+        reboot_timeout: "{{ reboot.update_timeout }}"
+
+    - name: Assert latest minor version
+      assert:
+        that: ansible_distribution_version == "{{ required_minor_version }}"
+        quiet: true
+
+    - name: Install packages that provide the leapp utility
+      package:
+        name: "{{ leapp_dependencies.packages[provider] }}"
+        state: present
+
+    # The following leapp_metadata archive is tied to a subscription,
+    # thus it has been encrypted to avoid licensing conflicts
+    - name: Create leapp metadata archive
+      copy:
+        content: "{{ metadata_archive | b64decode }}"
```

Comment:
I think we can provide this file in different way - without adding it to public GH repo. Maybe add it to Kubernetes on the build cluster?